be/src/vec/functions/array/function_array_distinct.h

```diff
@@ -0,0 +1,294 @@
+            res = _execute_number<UInt8>(src_column, src_offsets, dest_column, dest_offsets,
+                                         src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::UInt16) {
+            res = _execute_number<UInt16>(src_column, src_offsets, dest_column, dest_offsets,
+                                          src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::UInt32) {
+            res = _execute_number<UInt32>(src_column, src_offsets, dest_column, dest_offsets,
+                                          src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::UInt64) {
+            res = _execute_number<UInt64>(src_column, src_offsets, dest_column, dest_offsets,
+                                          src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::UInt128) {
+            res = _execute_number<UInt128>(src_column, src_offsets, dest_column, dest_offsets,
+                                           src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Int8) {
+            res = _execute_number<Int8>(src_column, src_offsets, dest_column, dest_offsets,
+                                        src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Int16) {
+            res = _execute_number<Int16>(src_column, src_offsets, dest_column, dest_offsets,
+                                         src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Int32) {
+            res = _execute_number<Int32>(src_column, src_offsets, dest_column, dest_offsets,
+                                         src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Int64) {
+            res = _execute_number<Int64>(src_column, src_offsets, dest_column, dest_offsets,
+                                         src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Int128) {
+            res = _execute_number<Int128>(src_column, src_offsets, dest_column, dest_offsets,
+                                          src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Float32) {
+            res = _execute_number<Float32>(src_column, src_offsets, dest_column, dest_offsets,
+                                           src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Float64) {
+            res = _execute_number<Float64>(src_column, src_offsets, dest_column, dest_offsets,
+                                           src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::Date) {
+            res = _execute_number<Date>(src_column, src_offsets, dest_column, dest_offsets,
+                                        src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::DateTime) {
```

Comment:
need add decimal support

Reply:
> need add decimal support

yes, I am adding function for the decimal.